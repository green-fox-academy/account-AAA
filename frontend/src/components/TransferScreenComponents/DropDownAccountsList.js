import React from 'react';
import PropTypes from 'prop-types';
import {
  Text, Icon, Picker, Card, CardItem,
} from 'native-base';

export default function DropDownAccountsList({ accounts, transferAccountSelection }) {
  const [selected, setSelected] = React.useState(undefined);

  const onValueChange = (value) => {
    setSelected(value);
    transferAccountSelection(value);
  };

  return (
    <Card>
      <CardItem>
        <Text>To: </Text>
        <Picker
          mode="dropdown"
          iosIcon={(
            <Icon
              name="arrow-dropdown-circle"
              style={{ color: '#007aff', fontSize: 25 }}
            />
          )}
          placeholder="Select Your Account"
          placeholderStyle={{ color: '#bfc6ea' }}
          placeholderIconColor="#007aff"
          style={{ width: undefined }}
          selectedValue={selected}
          onValueChange={(value) => onValueChange(value)}
        >
          {accounts.map((account) => (
            <Picker.Item
              key={account.id}
              label={account.depositName}
              value={account.depositName}
            />
          ))}
        </Picker>
      </CardItem>
    </Card>

  );
}

DropDownAccountsList.propTypes = {
  accounts: PropTypes.arrayOf(PropTypes.object).isRequired,
  transferAccountSelection: PropTypes.func.isRequired,
};
