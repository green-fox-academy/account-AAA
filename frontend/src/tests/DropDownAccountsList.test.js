import React from 'react';
import renderer from 'react-test-renderer';
import DropDownAccountsList from '../components/TransferScreenComponents/DropDownAccountsList';

describe('<DropDownAccountsList />', () => {
  it('render correctly', () => {
    const accounts = [{ id: 1, depositName: 'Main Account' }, { id: 2, depositName: 'Saving Account' }];
    const transferAccountSelection = () => null;
    const tree = renderer
      .create(
        <DropDownAccountsList
          accounts={accounts}
          transferAccountSelection={transferAccountSelection}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
