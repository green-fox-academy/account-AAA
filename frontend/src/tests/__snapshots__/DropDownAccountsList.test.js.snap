// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropDownAccountsList /> render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderBottomWidth": 0.5,
      "borderColor": "#ccc",
      "borderLeftWidth": 0.5,
      "borderRadius": 2,
      "borderRightWidth": 0.5,
      "borderTopWidth": 0.5,
      "elevation": 3,
      "flexWrap": "nowrap",
      "marginBottom": 5,
      "marginLeft": 2,
      "marginRight": 2,
      "marginTop": 5,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 1.5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    >
      To: 
    </Text>
    <View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
            "width": undefined,
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16.5,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              },
              undefined,
            ]
          }
          uppercase={false}
        >
          Main Account
        </Text>
        <Text />
      </View>
      <Modal
        animationType="slide"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        transparent={false}
        visible={false}
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
              "height": 1334,
            }
          }
        >
          <View
            onLayout={[Function]}
          >
            <View
              style={
                Object {
                  "backgroundColor": "#F8F8F8",
                  "borderBottomColor": "#a7a6ab",
                  "borderBottomWidth": 0.5,
                  "elevation": 3,
                  "flexDirection": "row",
                  "height": 64,
                  "justifyContent": "center",
                  "left": 0,
                  "paddingLeft": 6,
                  "paddingRight": 10,
                  "paddingTop": 18,
                  "right": 0,
                  "shadowColor": undefined,
                  "shadowOffset": undefined,
                  "shadowOpacity": undefined,
                  "shadowRadius": undefined,
                  "top": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "alignSelf": "center",
                    "flex": 1,
                  }
                }
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": null,
                      "backgroundColor": "transparent",
                      "borderBottomWidth": null,
                      "borderColor": null,
                      "borderLeftWidth": null,
                      "borderRadius": 5,
                      "borderRightWidth": null,
                      "borderTopWidth": null,
                      "borderWidth": undefined,
                      "elevation": 0,
                      "flexDirection": "row",
                      "height": 45,
                      "justifyContent": "space-between",
                      "marginLeft": 3,
                      "opacity": 1,
                      "paddingBottom": 6,
                      "paddingLeft": 4,
                      "paddingRight": 6,
                      "paddingTop": 6,
                      "shadowColor": null,
                      "shadowOffset": null,
                      "shadowOpacity": null,
                      "shadowRadius": null,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#007aff",
                        "fontFamily": "System",
                        "fontSize": 17,
                        "marginLeft": 0,
                        "marginRight": 0,
                        "paddingLeft": 2,
                        "paddingRight": undefined,
                        "top": 1,
                      }
                    }
                    uppercase={false}
                  >
                    Back
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "flex": 1,
                  }
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 17,
                      "fontWeight": "700",
                      "marginLeft": undefined,
                      "paddingLeft": 4,
                      "paddingTop": 1,
                      "textAlign": "center",
                    }
                  }
                >
                  Select One
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "flex-end",
                    "alignSelf": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                  }
                }
              />
            </View>
          </View>
          <RCTScrollView
            data={
              Array [
                <Item
                  label="Main Account"
                  value="Main Account"
                />,
                <Item
                  label="Saving Account"
                  value="Saving Account"
                />,
              ]
            }
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <View>
              <View
                onLayout={[Function]}
                style={null}
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={null}
                >
                  <View
                    button={true}
                    onPress={[Function]}
                    selected={true}
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "borderBottomWidth": 0.5,
                        "borderColor": "#c9c9c9",
                        "flexDirection": "row",
                        "marginLeft": 16,
                        "paddingBottom": 13,
                        "paddingRight": 16,
                        "paddingTop": 13,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "flex-start",
                          "alignSelf": "center",
                          "flex": 1,
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "alignSelf": "center",
                            "color": "#007aff",
                            "fontFamily": "System",
                            "fontSize": 16,
                          }
                        }
                        uppercase={false}
                      >
                        Main Account
                      </Text>
                    </View>
                    <View
                      style={
                        Object {
                          "alignItems": "flex-end",
                          "alignSelf": "center",
                          "flex": 0.28,
                          "paddingBottom": null,
                          "paddingLeft": null,
                          "paddingRight": null,
                          "paddingTop": null,
                        }
                      }
                    >
                      <View
                        accessible={true}
                        isTVSelectable={true}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignSelf": null,
                            "opacity": 1,
                          }
                        }
                      >
                        <Text />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                onLayout={[Function]}
                style={null}
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={null}
                >
                  <View
                    button={true}
                    onPress={[Function]}
                    selected={false}
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "borderBottomWidth": 0.5,
                        "borderColor": "#c9c9c9",
                        "flexDirection": "row",
                        "marginLeft": 16,
                        "paddingBottom": 13,
                        "paddingRight": 16,
                        "paddingTop": 13,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "flex-start",
                          "alignSelf": "center",
                          "flex": 1,
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "alignSelf": "center",
                            "color": "#000",
                            "fontFamily": "System",
                            "fontSize": 16,
                          }
                        }
                        uppercase={false}
                      >
                        Saving Account
                      </Text>
                    </View>
                    <View
                      style={
                        Object {
                          "alignItems": "flex-end",
                          "alignSelf": "center",
                          "flex": 0.28,
                          "paddingBottom": null,
                          "paddingLeft": null,
                          "paddingRight": null,
                          "paddingTop": null,
                        }
                      }
                    >
                      <View
                        accessible={true}
                        isTVSelectable={true}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignSelf": null,
                            "opacity": 1,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </Modal>
    </View>
  </View>
</View>
`;
